nil pointer dereference when running `tailscale netcheck`

### What is the issue?

Running `tailscale netcheck` while connected to an exit node causes a nil pointer deref.

Client: Tailscale 1.32.0 on Windows 11 (22H2)
Exit Node: Tailscale 1.32.0 on Windows 10 (21H2)

### Steps to reproduce

```
C:\Users\Nebula>tailscale netcheck
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1378238]

goroutine 175 [running]:
tailscale.com/derp/derphttp.(*Client).Close(0xc00069c2d0)
        tailscale.com@v1.32.0/derp/derphttp/derphttp_client.go:988 +0x38
tailscale.com/net/netcheck.(*Client).measureHTTPSLatency(0xc000240000, {0x15f3668, 0xc0002802a0}, 0x0?)
        tailscale.com@v1.32.0/net/netcheck/netcheck.go:1195 +0xa7c
tailscale.com/net/netcheck.(*Client).GetReport.func8(0xc0002b6640)
        tailscale.com@v1.32.0/net/netcheck/netcheck.go:1033 +0x97
created by tailscale.com/net/netcheck.(*Client).GetReport
        tailscale.com@v1.32.0/net/netcheck/netcheck.go:1031 +0x1c72
```

### Are there any recent changes that introduced the issue?

_No response_

### OS

Windows

### OS version

22H2

### Tailscale version

1.32.0

### Bug report

BUG-a01c06d9d3e148a6c8c549f5ac2362b66b80190e0c272eed365057a70efa314f-20221013140405Z-6b8508d4502d0499

--- a/derp/derphttp/derphttp_client.go
+++ b/derp/derphttp/derphttp_client.go
@@ -96,7 +96,7 @@ func NewRegionClient(privateKey key.NodePrivate, logf logger.Logf, getRegion fun
 	return c
 }
 
-// NewNetcheckClient returns a Client that's only able to have its DialRegion method called.
+// NewNetcheckClient returns a Client that's only able to have its DialRegionTLS method called.
 // It's used by the netcheck package.
 func NewNetcheckClient(logf logger.Logf) *Client {
 	return &Client{logf: logf}
@@ -985,7 +985,9 @@ func (c *Client) isClosed() bool {
 // Close closes the client. It will not automatically reconnect after
 // being closed.
 func (c *Client) Close() error {
-	c.cancelCtx() // not in lock, so it can cancel Connect, which holds mu
+	if c.cancelCtx != nil {
+		c.cancelCtx() // not in lock, so it can cancel Connect, which holds mu
+	}
 
 	c.mu.Lock()
 	defer c.mu.Unlock()

